File: day_1/puzzle.py
left = []
right = []

file = open("puzzle.input")
for line in file.readlines():
    nr = line.strip().split("   ")
    n1 = int(nr[0])
    n2 = int(nr[1])
    left.append(n1)
    right.append(n2)

left.sort()
right.sort()

# Part 1
s1 = 0
for i in range(0, len(left)):
    a = left[i]
    b = right[i]
    s1 += abs(a-b)

print("Part 1: " + str(s1))

# Part 2
s2 = 0
for a in left:
    cnt = right.count(a)
    s2 += a * cnt 

print("Part 2: " + str(s2))

File: day_1/puzzle.rs
use std::fs::File;
use std::vec::Vec;
use std::io::{self, BufRead};

fn main() {
    let mut left: Vec<i32> = Vec::new();
    let mut right: Vec<i32> = Vec::new();
    
    let file = File::open("puzzle.input").unwrap();
    let reader = io::BufReader::new(file);

    for line in reader.lines() {
        let line = line.unwrap();
        let nr: Vec<&str> = line.split_whitespace().collect();
        let n1 = nr[0].parse::<i32>().unwrap();
        let n2 = nr[1].parse::<i32>().unwrap();
        left.push(n1);
        right.push(n2);
    }

    left.sort();
    right.sort();

    // Paer 1
    let mut s1: i32 = 0;
    for i in 0..left.len() {
        let a = left.get(i).unwrap();
        let b = right.get(i).unwrap();

        s1 += (*a - *b).abs();
    }

    println!("Part 1: {}", s1);

    // Part 2
    let mut s2: i32 = 0;
    for a in left {
        let cnt = right.iter().filter(|&i| *i == a).count();
        s2 += cnt as i32 * a;
    }

    println!("Part 2: {}", s2);
}
